runtime/compiler/exceptions/AOTFailure.hpp

```diff
@@ -90,6 +90,16 @@ class AOTHasMethodTypeConstant : public virtual TR::RecoverableILGenException
    virtual const char* what() const throw() { return "AOT Has Method Type Constant"; }
    };
 
+/**
+ * AOT Has patched CP constant exception type.
+ *
+ * Thrown when a method that has a constant object in CP entry patched to a different type is AOT Compiled.
+ */
+class AOTHasPatchedCPConstant: public virtual TR::RecoverableILGenException
```

Comment:
Question - should this be recoverable? We can recover only with a non-AOT compile so wanted ot make sure this is the right semantics

Reply:
Yes.  Added code to disable AOT in recompilation.